Added docs for submitting and working tickets. (#8)
Python-Practice-Discord/Project-Template: docs/TICKETS.md

# Tickets
This project uses of GitHub Issues for tracking and managing tasks.

# Submitting Tickets
Submitting tickets is an easy way to contribute to the project, with no programming skills required.
When submitting a ticket, please make sure to include as much information as you can,
so a contributor can best work off of it.

Before submitting a ticket, please search through existing tickets to make sure that you're not
submitting a duplicate. Duplicate tickets will be closed.

## Bug Reports and Feature Requests
If you discover a bug in the project, or you would like to have a feature added to it,
please submit a bug report or a feature request.

To submit a bug report or feature request, go to the "Issues" tab in the repository,
and click "New Issue". Then, select the Bug Report template or the feature request template.

Make sure you include as much of the requested information as you have, since it makes it much
easier for contributors to investigate.

# Working a Ticket
Members of the Core team, and the Team Lead should "work" each ticket submitted, by applying the
proper labels, gathering information, answering questions, and eventually, implementing new

> features, and eventually, closing the ticket.

Reviewer: ```suggestion
Members of the Core team, and the Team Lead should "work" each ticket submitted, by applying the
proper labels, gathering information, answering questions, implementing new
features, and eventually, closing the ticket.
```

## Labels
It is important that each new ticket has the correct labels applied to it. This makes it easier

> for contributors to quickly search through for related tickets, or for things to work on.

Reviewer: ```suggestion
It is important that each new ticket has the correct labels applied to it. This makes it easier
for contributors to quickly search through for related tickets, or find issues to work on.
```

This project's ticket labels are as follows:
- `documentation`
    - Requests for improvements and/or additions to the project's documentation.
- `enhancement`
    - Feature Requests
- `bug`
    - Bug reports
- `question`
    - General questions about the project.
- `duplicate`
    - Tickets that are closely related to or duplicates of another ticket.
- `invalid`
    - A ticket that contains generally invalid or false information, like a bug report that can't
      be validated, a feature request for a feature that already exists.
- `triageed`

>     - Tickets that have been confirmed to be implemented into the project.

Reviewer: ```suggestion
    - Tickets that have been confirmed and will be implemented into the project.
```

    - Only the Team Lead should apply the `triaged` label to a ticket.
- `help wanted`
    - Tickets that can be picked up by a contributor outside the core team.
    - Only the Team Lead should apply the `help wanted` label to a ticket.
- `good first issue`
    - Tickets that are easy for a new contributor to pick up and implement.
    - A `good first issue` ticket is detailed and easy to implement, and should have a detailed
      explanation of what needs to be done, so that a new contributor can easily pick it up.
    - Only the Team Lead should apply the `good first issue` label to a ticket.
- `wontfix`
    - Tickets that contain enhancements or bug reports that will not be accepted.
      and added to the project at all.
    - Only the Team Lead should apply the `wontfix` label to a ticket.

## Assigning and Picking Up Tickets
### Core Team Members:
Once a ticket is marked as `triaged`, if you'd like to pick up the ticket to work on implementing it,

> you can assign themselves to the ticket. Do this by clicking the 'assignees' button at the top right,

Reviewer: ```suggestion
you can assign yourself to the ticket. Do this by clicking the 'assignees' button at the top right,
```

and selecting yourself.

Make sure that nobody has already assigned themselves before assigning yourself or beginning work,
so that you don't waste your time with a duplicate pull request.

### Outside Contributors
If a ticket is marked as `help wanted` or `good first issue`, and you'd like to pick up the ticket to
work on implementing it, post a comment in the ticket saying that you'll be working on it. When a
Core Team member sees it, they will mark it as assigned to you.

Make sure that nobody has asked to be assigned or already been assigned before you begin working,
to prevent multiple people opening a pull request over the same thing.

## Closing Tickets

> Tickets should be closed under the issues raised are resolved. This means:

Reviewer: ```suggestion
The criteria for ticket closure is as follows:
```
